src/librustdoc/clean/mod.rs

```diff
@@ -469,6 +469,7 @@ fn clean_projection<'tcx>(
 
     let trait_ =
         clean_trait_ref_with_bindings(cx, ty.map_bound(|ty| ty.trait_ref(cx.tcx)), ThinVec::new());
+    // FIXME: Maybe add comment here why `None` is fine.
```

Comment:
Well, please do. :D

Reply:
For posterity's sake, if/once we respect object lifetime defaults of GATs, `None` would be fine here since `Self` doesn't have a meaningful object lifetime default according to

https://github.com/rust-lang/rust/blob/b30b535124aabf7835a09fe1c07e469711964011/compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs#L1563-L1564

If #115379 (the 1000th time I'm mentioning it lol) is indeed a bug, then I will come back to this an add a comment :)